src/sql_parser.rs

```diff
@@ -0,0 +1,175 @@
+//! SQL parser utilities for extracting metadata from SQL statements.
+//!
+//! This module provides lightweight SQL parsing to extract information
+//! needed for optimizations, such as the table name from INSERT statements.
+
+use sqlparser::ast::{ObjectName, ObjectNamePart, Statement, TableObject};
+use sqlparser::dialect::GenericDialect;
+use sqlparser::parser::Parser;
+
+/// Represents a parsed SQL statement with extracted metadata.
+///
+/// This provides an object-oriented interface for working with parsed SQL,
+/// making it easier to extend for DQ statements and other use cases.
+pub struct ParsedStatement {
+    statement: Statement,
+}
+
+impl ParsedStatement {
+    /// Parse a SQL statement.
+    ///
+    /// Returns `None` if the SQL cannot be parsed or contains multiple statements.
+    pub fn parse(sql: &str) -> Option<Self> {
+        let dialect = GenericDialect {};
+        let statements = Parser::parse_sql(&dialect, sql).ok()?;
+
+        if statements.len() != 1 {
+            return None;
+        }
+
+        Some(Self {
+            statement: statements.into_iter().next()?,
+        })
+    }
+
+    /// Check if this is an INSERT statement.
+    pub fn is_insert(&self) -> bool {
+        matches!(self.statement, Statement::Insert(_))
+    }
+
+    /// Check if this is a query statement (returns results).
+    ///
+    /// Returns true for SELECT, SHOW, EXPLAIN, and other query statements.
+    /// Returns false for INSERT, UPDATE, DELETE, CREATE, ALTER, DROP, etc.
+    pub fn is_query(&self) -> bool {
+        matches!(
+            self.statement,
+            Statement::Query(_)
+                | Statement::ShowTables { .. }
+                | Statement::ShowColumns { .. }
+                | Statement::ShowCreate { .. }
+                | Statement::ShowVariable { .. }
+                | Statement::ShowVariables { .. }
+                | Statement::Explain { .. }
+                | Statement::ExplainTable { .. }
+        )
```

Comment:
I see in our delete code, we also set `PRAGMA` as query, have we handle `PRAGMA` here?